library/bignum_core.h

```diff
@@ -109,7 +109,38 @@
  * \return      The number of leading zero bits in \p a, if \p a != 0.
  *              If \p a == 0, the result is undefined.
  */
-size_t mbedtls_mpi_core_clz(mbedtls_mpi_uint a);
+static inline size_t mbedtls_mpi_core_clz(mbedtls_mpi_uint a)
```

Comment:
It's weird that arranging to have two copies of the function (since it's used in two different C files) makes the code smaller. How is this happening? Is this maybe only on platforms that have a CLZ instruction — what's the impact on platforms that don't?

Reply:
Without `clz` there’s still a 16-byte saving in `bignum_core.o`, it's just that it is 20 bytes bigger without `clz` in both `development` and this PR (code size goes from 2446 to 2466 without `clz`)

Reply:
Ah - that's just in the TF-M config (with lots of things turned off) - so in my measurements it's not being pulled into bignum.o

Reply:
I suspect a clz instruction is near-universal. E.g., Arm has had it since Armv5. So this function will almost always turn into a single instruction.

Reply:
But actually, I'm not sure the inline is useful. Removing it completely results in an inlined `clz` instruction, and `mbedtls_mpi_core_clz` is not present in `libmbedcrypto.a`. So I'm going to remove it, it seems the compiler does the right thing without it.

Reply:
It seems to get inlined regardless of marking it `static inline` (at least on my machine where the `clz` instruction is present). So the effect of the `static inline` is really just to eliminate the not-inlined copy of the function, which is small and presumably eliminated during the final link stage. Do we want to simplify and drop the inlining?

Reply:
This PR saved bytes over the code in development. I think it is okay

Reply:
> should we return it to bignum_core.c, as a non-inline, non-static function, and retain a declaration only in the header?

Yes, we should do the simple thing (declare in header, define in .c) unless there's a compelling reason to do otherwise. And it seems that there's no compelling reason after all.